### DISCUSSION TITLE
Multi Job/Multi Device/File configuration

### DISCUSSION THREAD

**Discussion by levelrage** [louwrentius/fio-plot#143] [General]
First I'll say I am just getting familiar with the code base and I am digging this fio wrapper and plotter.  

My primary question at this point is there a way to not just run in 'parallel' but to run as 'combined'.

First it appears base on my first test of '--parallel' is that it runs multiple individuals instances of fio instead of running a single 'instance' of fio.
        * I.E. ./fio <global params> --name test1 --filename=/dev/nvme0n1 & ./fio <global params> --name test2 --filename=/dev/nvme1n1 & ./fio <global params> --name test3 --filename=/dev/nvme2n1

As for 'combined' I imagine them in two different methods:
1. Use the global configuration area to layout each jobs configuration including the '--group_reporting' parameter and then use multiple '--name' and '--filename' in order to define multiple 'jobs' to benchmark multiple file and or devices within a single reporting command.
        * This essentially duplicates the global parameters to a per job/file/device basis but runs them all in parallel .
        * I.E. ./fio <global params> --name test1 --filename=/dev/nvme0n1 <TEST1 params> --name test2 --filename=/dev/nvme1n1 <TEST2 params> --name test3 --filename=/dev/nvme2n1 <TEST3 params>
2. Instead of using multiple '--name' and '--filename' you simply list devices/files in a ':' delimited list to '--filename'  This again allows for multi device/file testing but essentially is a single 'job' based on the global test parameters
        * I.E. ./fio <global params> --name test1 --filename=/dev/nvme0:/dev/nvme1:/dev/nvme2

Sorry if I fundamentally missed something.

**Comment by louwrentius** [comment]
Hi sorry for the late reply. 

As I understand your question, your main goal seems to be that the benchmark script should output a single output file with benchmark data combined. The output will be a JSON that contains the output of multiple jobs (each device is a job). I guess this is also related to fio group_reporting and stuff. 

Is this correct?

**Reply by levelrage** [reply]
Essentially, although I'd say it is more about a single 'fio' command that tests multiple devices on the same host or on multiple hosts.

As mentioned there are two methods of performing this.  

1. '--filename=' with multiple devices or files listed in a ':' delimited list.
2. Multiple devices specified in concurrent jobs by designating each file or device with a '--name=test1 --filename=/dev/nvme0n1'

**Reply by louwrentius** [reply]
Yes I understand, but I also want to understand what your goal is: why is this important to you?

**Reply by levelrage** [reply]
Sure thing!  Just fairly common testing methodology into the enterprise storage space.
